test/functional/test-standard-actions.js

```diff
@@ -419,6 +419,94 @@ describes.sandboxed('StandardActions', {}, () => {
       });
     });
 
+    describe('closeOrNavigateTo', () => {
+      let navigateToStub;
+      let closeOrNavigateToSpy;
+      let winCloseSpy;
+
+      beforeEach(() => {
+        navigateToStub = sandbox.stub(standardActions, 'handleNavigateTo')
+            .returns(Promise.resolve());
+
+        closeOrNavigateToSpy = sandbox.spy(standardActions,
+            'handleCloseOrNavigateTo');
+
+        win.close = () => {
```

Comment:
It isn't quite clear here if there was a `win.close` before, and it wouldn't get reset after the test if there was. If there is already a method, you should do:

```javascript
winCloseStub = sandbox.stub(win, 'close');
```

If there is no `win.close` already, you should do something like:

```javascript
winCloseStub = win.close = sinon.stub(win);
```

You can then control what it does with:

```javascript
winCloseStub.callsFake(() => {
  win.closed = true;
});
```

You can also control what it does on each call, which is more powerful than simply defining the method. For example:

```javascript
winCloseStub.onFirstCall().callsFake(() => {
  win.closed = true;
});
winCloseStub.onSecondCall().callsFake(() => {
  win.closed = false;
});
```

for example, you should be able to change the behavior of `win.close` for the last test without losing the stub/spy you set up here.